Service Template not working with OpenNebula > 5.12

`TestAccServiceTemplate` test is failing when testing the provider against OpenNebula version > 5.12:

```
=== RUN   TestAccServiceTemplate
    testing.go:715: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: errors during apply: <!DOCTYPE html>

...

--- FAIL: TestAccServiceTemplate (0.32s)
=== RUN   TestAccService
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc0 pc=0xfb42a7]
```

The complete error can be found [here](https://github.com/OpenNebula/terraform-provider-opennebula/runs/4140792742?check_suite_focus=true).

--- a/.ci_scripts/oned.sh
+++ b/.ci_scripts/oned.sh
@@ -21,10 +21,14 @@ mkdir ~/.one
 echo 'oneadmin:opennebula' > ~/.one/one_auth
 
 # Enable dummy drivers
-sudo chmod o+w /etc/one/oned.conf
+sudo chmod o+wr /etc/one/oned.conf
 sudo chmod o+w /etc/one/monitord.conf
+sudo chmod 777 /etc/one
+sed '/DATASTORE_MAD/,+3 d' -i /etc/one/oned.conf
 echo 'IM_MAD = [ NAME="dummy", SUNSTONE_NAME="Dummy", EXECUTABLE="one_im_sh", ARGUMENTS="-r 3 -t 15 -w 90 dummy", THREADS=0]' >> /etc/one/monitord.conf
 echo 'VM_MAD = [ NAME="dummy", SUNSTONE_NAME="Testing", EXECUTABLE="one_vmm_dummy",TYPE="xml" ]' >> /etc/one/oned.conf
+echo 'DATASTORE_MAD = [ EXECUTABLE = "one_datastore", ARGUMENTS  = "-t 15 -d dummy,fs,lvm,ceph,dev,iscsi_libvirt,vcenter -s dummy,shared,ssh,ceph,fs_lvm,fs_lvm_ssh,qcow2,vcenter"]' >> /etc/one/oned.conf
+
 
 # start oned
 sudo systemctl start opennebula
@@ -36,3 +40,16 @@ timeout 60 sh -c 'until nc -z $0 $1; do sleep 1; done' localhost 2633
 # Create dummy host
 sudo su -c "onehost create dummy -i dummy -v dummy" oneadmin
 
+# Update DS drivers
+tmp_file=$(mktemp)
+
+echo "TM_MAD=\"dummy\"" > "$tmp_file"
+onedatastore update 0 "$tmp_file"
+echo "DS_MAD=\"dummy\"" >> "$tmp_file"
+onedatastore update 1 "$tmp_file"
+
+# Force DS monitoring
+sudo systemctl restart opennebula
+
+# check it's up
+timeout 60 sh -c 'until nc -z $0 $1; do sleep 1; done' localhost 2633

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,7 @@ BUG FIXES:
 * resources/opennebula_security_group: fix rule conversion from struct to config
 * resources/opennebula_virtual_machine_group: make `role` reading conditional
 * resources/opennebula_virtual_machine_group: remove `vms` field
+* resources/opennebula_service: add compatibility with OneFlow server > `5.12.x`
 * data/opennebula_user: remove password field
 
 FEATURES:

--- a/opennebula/resource_opennebula_service.go
+++ b/opennebula/resource_opennebula_service.go
@@ -197,6 +197,15 @@ func resourceOpennebulaServiceCreate(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
+	_, err = waitForServiceState(d, meta, "running")
+	if err != nil {
+		service, _ := sc.Info()
+
+		svState := service.Template.Body.StateRaw
+		return fmt.Errorf(
+			"Error waiting for Service (%s) to be in state RUNNING: %s (state: %v)", d.Id(), err, svState)
+	}
+
 	return resourceOpennebulaServiceRead(d, meta)
 }
 
@@ -468,7 +477,7 @@ func waitForServiceState(d *schema.ResourceData, meta interface{}, state string)
 		return service, err
 	}
 
-	log.Printf("Waiting for Service (%s) to be in state Done", d.Id())
+	log.Printf("Waiting for Service (%s) to be in state %s", d.Id(), state)
 
 	stateConf := &resource.StateChangeConf{
 		Pending: []string{"anythingelse"}, Target: []string{state},
@@ -485,7 +494,11 @@ func waitForServiceState(d *schema.ResourceData, meta interface{}, state string)
 			service, err = sc.Info()
 			if err != nil {
 				if strings.Contains(err.Error(), "Error getting") {
-					return service, "notfound", nil
+					if state == "done" {
+						return service, "done", nil // DONE == notfound for ONE > 5.12
+					} else {
+						return service, "notfound", nil
+					}
 				}
 				return service, "", err
 			}

